www/src/components/repos/common/misc.js

```diff
@@ -93,49 +96,84 @@ export function PackageHeader({ name, icon }) {
   )
 }
 
-// TODO: Implement view more functionality as at the moment it loads only the first page.
-// TODO: Show only 6 elements inside select component and add scroll (to be done globally).
 export function PackageVersionPicker({
-  edges, installed, version, setVersion,
+  edges, installed, version, setVersion, pageInfo, fetchMore,
 }) {
+  const versions = edges.map(({ node }) => node)
+
   return (
     <Box
+      width="240px"
       gap="small"
       margin={{ bottom: 'medium' }}
     >
       <Select
-        value={version}
-        onChange={({ target: { value } }) => setVersion(value)}
-        style={{ maxWidth: '240px' }}
+        label="version"
+        width="240px"
+        selectedKey={version.id}
+        onSelectionChange={selected => setVersion(versions.find(v => v.id === selected))}
+        dropdownFooter={pageInfo.hasNextPage && <ListBoxFooterPlus>View more</ListBoxFooterPlus>}
+        onFooterClick={() => fetchMore({
+          variables: { cursor: pageInfo.endCursor },
+          updateQuery: (prev, { fetchMoreResult: { versions } }) => extendConnection(prev, versions, 'versions'),
+        })}
+        rightContent={
+          version.id === installed?.version?.id && (
+            <ListBoxItemChipList chips={[
+              <Chip
+                severity="success"
+                size="small"
+              >
+                Installed
+              </Chip>,
+            ]}
+            />
+          )
+        }
       >
-        {edges.map(({ node }) => (
-          <MenuItem
-            key={node.id}
-            value={node}
-          >
-            <Box
-              fill
-              direction="row"
-              justify="center"
-            >
-              <Box fill>{node.version}</Box>
-              {node.id === installed?.version?.id && (
-                <Chip
-                  severity="success"
-                  size="small"
-                >
-                  Installed
-                </Chip>
-              )}
-            </Box>
-          </MenuItem>
+        {versions.map(v => (
+          <ListBoxItem
+            key={v.id}
+            label={v.version}
+            rightContent={(
+              <ListBoxItemChipList
+                maxVisible={1}
+                showExtra
+                chips={[
+                  v.id === installed?.version?.id && (
+                    <Chip
+                      severity="success"
+                      size="small"
+                    >
+                      Installed
+                    </Chip>
+                  ),
```

Comment:
This inserts `false` as the first element of the array if it's not the installed version. So the wrong number of chips gets displayed.

Reply:
This should work: `...(v.id === installed?.version?.id ? [(<Chip />)] : [])`

Reply:
Good catch, thanks.